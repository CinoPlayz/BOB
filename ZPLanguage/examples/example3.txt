infrastructure "Celje-Maribor" {
    let celjePostaja = station "Celje" {
        box((3,2), (5,3));
        platform 1, 2 {
            box((4,2), (2,3));
        }
        platform 2, 2 {
            box((6,3), (3,5));
        }
    }

    let mariborPostaja = station "Maribor" {
        box((6,4), (5,3));
        platform 1, 2 {
            box((5,2), (4,3));
        }
    }

     let celjeTir = track "Celje-Maribor", celjePostaja->1->1, mariborPostaja->1->2 {
        line(null, (2,5));
        line((2,4), null);
    }

     let mariborTir = track "Maribor-Celje", mariborPostaja->1->1, celjePostaja->2->1{
            line(null, (4,3));
            line(null, (5, 7));
            line(null, null);
     }

    switch celjeTir, mariborTir {
        line((3,2), (4,2));
    }

    switch mariborTir, celjeTir {
    }
}