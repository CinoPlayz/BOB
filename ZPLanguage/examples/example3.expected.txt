{
                  "type": "FeatureCollection",
                  "features": [                {
                  "type": "Feature",
                  "properties": {
                    "type": "station",
                    "name": "Celje"
                  },
                                      "geometry": {
   "type": "Polygon",
   "coordinates": [[
      [2.0, 3.0],
      [3.0, 3.0],
      [3.0, 5.0],
      [2.0, 5.0],
      [2.0, 3.0]
   ]]
}
                },                {
                  "type": "Feature",
                  "properties": {
                    "type": "platform",
                    "number": "1",
                    "countOfTracks": "2",
                    "stationName": "Celje"
                  },
                                      "geometry": {
   "type": "Polygon",
   "coordinates": [[
      [2.0, 4.0],
      [3.0, 4.0],
      [3.0, 2.0],
      [2.0, 2.0],
      [2.0, 4.0]
   ]]
}
                },                {
                  "type": "Feature",
                  "properties": {
                    "type": "platform",
                    "number": "2",
                    "countOfTracks": "2",
                    "stationName": "Celje"
                  },
                                      "geometry": {
   "type": "Polygon",
   "coordinates": [[
      [3.0, 6.0],
      [5.0, 6.0],
      [5.0, 3.0],
      [3.0, 3.0],
      [3.0, 6.0]
   ]]
}
                },                {
                  "type": "Feature",
                  "properties": {
                    "type": "station",
                    "name": "Maribor"
                  },
                                      "geometry": {
   "type": "Polygon",
   "coordinates": [[
      [4.0, 6.0],
      [3.0, 6.0],
      [3.0, 5.0],
      [4.0, 5.0],
      [4.0, 6.0]
   ]]
}
                },                {
                  "type": "Feature",
                  "properties": {
                    "type": "platform",
                    "number": "1",
                    "countOfTracks": "2",
                    "stationName": "Maribor"
                  },
                                      "geometry": {
   "type": "Polygon",
   "coordinates": [[
      [2.0, 5.0],
      [3.0, 5.0],
      [3.0, 4.0],
      [2.0, 4.0],
      [2.0, 5.0]
   ]]
}
                },                {
                  "type": "Feature",
                  "properties": {
                    "type": "track",
                    "name": "Celje-Maribor",
                    "startPlatformStationName": "Celje",
                    "startPlatformNumber": "1",
                    "startPlatformCountOfTracks": "2",
                    "endPlatformStationName": "Maribor",
                    "endPlatformNumber": "1",
                    "endPlatformCountOfTracks": "2"
                  },

                    "geometry": {
                        "type": "GeometryCollection",
                        "geometries": [{
   "type": "LineString",
   "coordinates": [
      [2.2, 4.2],
      [5.0, 2.0]
   ]
}                     ,{
   "type": "LineString",
   "coordinates": [
      [4.0, 2.0],
      [1.8, 4.8]
   ]
}                     ]
                    }

                },                {
                  "type": "Feature",
                  "properties": {
                    "type": "track",
                    "name": "Maribor-Celje",
                    "startPlatformStationName": "Maribor",
                    "startPlatformNumber": "1",
                    "startPlatformCountOfTracks": "2",
                    "endPlatformStationName": "Celje",
                    "endPlatformNumber": "2",
                    "endPlatformCountOfTracks": "2"
                  },

                    "geometry": {
                        "type": "GeometryCollection",
                        "geometries": [{
   "type": "LineString",
   "coordinates": [
      [2.2, 5.2],
      [3.0, 4.0]
   ]
}                     ,{
   "type": "LineString",
   "coordinates": [
      [3.0, 4.0],
      [7.0, 5.0]
   ]
}                     ,{
   "type": "LineString",
   "coordinates": [
      [7.0, 5.0],
      [3.2, 6.2]
   ]
}                     ]
                    }

                },                {
                  "type": "Feature",
                  "properties": {
                    "type": "switch",
                    "trackOneName": "Celje-Maribor",
                    "trackTwoName": "Maribor-Celje"
                  },
                                      "geometry": {
   "type": "LineString",
   "coordinates": [
      [2.0, 3.0],
      [2.0, 4.0]
   ]
}
                },{
  "type": "Feature",
  "properties": {
    "type": "switch",
    "trackOneName": "Maribor-Celje",
    "trackTwoName": "Celje-Maribor"
  },

    "geometry": {
        "type": "GeometryCollection",
        "geometries": []
    }

}],
                  "metadata": {
                    "name": "Celje-Maribor"
                  }
                }